src/customers/components/CustomerListPage/CustomerListPage.tsx

```diff
@@ -6,54 +6,46 @@ import {
   useExtensions,
 } from "@dashboard/apps/hooks/useExtensions";
 import { useUserPermissions } from "@dashboard/auth/hooks/useUserPermissions";
+import { ListFilters } from "@dashboard/components/AppLayout/ListFilters";
 import { TopNav } from "@dashboard/components/AppLayout/TopNav";
-import ButtonWithSelect from "@dashboard/components/ButtonWithSelect";
-import CardMenu from "@dashboard/components/CardMenu/CardMenu";
-import FilterBar from "@dashboard/components/FilterBar";
+import { ButtonWithDropdown } from "@dashboard/components/ButtonWithDropdown";
+import { FilterPresetsSelect } from "@dashboard/components/FilterPresetsSelect";
+import { Customers } from "@dashboard/customers/types";
 import {
   customerAddUrl,
   CustomerListUrlSortField,
+  customerUrl,
 } from "@dashboard/customers/urls";
-import { ListCustomersQuery } from "@dashboard/graphql";
 import useNavigator from "@dashboard/hooks/useNavigator";
 import { sectionNames } from "@dashboard/intl";
-import {
-  FilterPageProps,
-  ListActions,
-  PageListProps,
-  RelayToFlat,
-  SortPage,
-  TabPageProps,
-} from "@dashboard/types";
-import { Card } from "@material-ui/core";
-import { makeStyles } from "@saleor/macaw-ui";
-import React from "react";
+import { FilterPageProps, PageListProps, SortPage } from "@dashboard/types";
+import { Box, Button, ChevronRightIcon } from "@saleor/macaw-ui/next";
+import React, { useState } from "react";
 import { FormattedMessage, useIntl } from "react-intl";
 
-import CustomerList from "../CustomerList/CustomerList";
+import { CustomerListDatagrid } from "../CustomerListDatagrid/CustomerListDatagrid";
+import { CustomerListDeleteButton } from "../CustomerListDeleteButton";
 import {
   createFilterStructure,
   CustomerFilterKeys,
   CustomerListFilterOpts,
 } from "./filters";
 
-const useStyles = makeStyles(
-  theme => ({
-    settings: {
-      marginRight: theme.spacing(2),
-    },
-  }),
-  { name: "CustomerListPage" },
-);
-
 export interface CustomerListPageProps
   extends PageListProps,
-    ListActions,
-    FilterPageProps<CustomerFilterKeys, CustomerListFilterOpts>,
-    SortPage<CustomerListUrlSortField>,
-    TabPageProps {
-  customers: RelayToFlat<ListCustomersQuery["customers"]>;
+    Omit<
+      FilterPageProps<CustomerFilterKeys, CustomerListFilterOpts>,
+      "onTabDelete"
+    >,
+    SortPage<CustomerListUrlSortField> {
+  customers: Customers | undefined;
   selectedCustomerIds: string[];
+  onTabUpdate: (tabName: string) => void;
+  loading: boolean;
+  hasPresetsChange: () => boolean;
+  onSelectCustomerIds: (rows: number[], clearSelection: () => void) => void;
+  onCustomersDelete: () => void;
+  onTabDelete: (id: number) => void;
```

Comment:
suggestion: I'd avoid using old name like `onTabUpdate`, I've create special interface for filter presets
Here is example from my PR with datagrid on discount page
```
export interface SaleListPageProps
  extends PageListProps,
    FilterPagePropsWithPresets<SaleFilterKeys, SaleListFilterOpts>,
    SortPage<SaleListUrlSortField>,
    ChannelProps {
  sales: SaleFragment[];
  selectedSaleIds: string[];
  onSalesDelete: () => void;
  onSelectSaleIds: (rows: number[], clearSelection: () => void) => void;
  ```